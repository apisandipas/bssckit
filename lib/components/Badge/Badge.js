"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _base = require("../Utilities/base");

var _themeFunctions = require("../../utils/themeFunctions");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n  display: inline-block;\n  font-size: 75%;\n  font-weight: ", ";\n  line-height: 1;\n  text-align: center;\n  white-space: nowrap;\n  vertical-align: baseline;\n  position: relative;\n  top: -1px;\n  &:empty {\n    display: none;\n  };\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n    & > a {\n      color: ", ";\n      text-decoration: none;\n      background-color: ", ";\n    }\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n      & > a {\n        color: ", ";\n        text-decoration: none;\n        background-color: ", ";\n        &:hover,\n        &:focus {\n          text-decoration: none;\n          background-color: ", ";\n        }\n      }\n      &:hover,\n      &:focus {\n        cursor: pointer;\n        background-color: ", ";\n        & > a {\n          background-color: ", ";\n        }\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", ";\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    padding: ", ";\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n      padding: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    border-radius: ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var borderRadius = function borderRadius(props) {
  if (props.pill) {
    return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getBorderRadius)(props, 'badge', 'pill'));
  } else if (props.noRadius) {
    return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getBorderRadius)(props, 'badge', 'noRadius'));
  }

  return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getBorderRadius)(props, 'badge', 'default'));
};

var padding = function padding(props) {
  if (props.pill) {
    return (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getPadding)(props, 'badge', 'pill'));
  }

  return (0, _styledComponents.css)(_templateObject5(), (0, _themeFunctions.getPadding)(props, 'badge', 'default'));
};

var backgroundColor = function backgroundColor(props) {
  return (0, _styledComponents.css)(_templateObject6(), (0, _themeFunctions.getColor)(props, 'badge', 'backgroundColor'));
};

var color = function color(props) {
  return (0, _styledComponents.css)(_templateObject7(), (0, _themeFunctions.getColor)(props, 'badge', 'color'));
};

var badgeLink = function badgeLink(props) {
  if (props.action) {
    return (0, _styledComponents.css)(_templateObject8(), (0, _themeFunctions.getColor)(props, 'badge', 'linkColor'), (0, _themeFunctions.getColor)(props, 'badge', 'backgroundColor'), (0, _themeFunctions.getColor)(props, 'badge', 'backgroundColorHoverFocus'), (0, _themeFunctions.getColor)(props, 'badge', 'backgroundColorHoverFocus'), (0, _themeFunctions.getColor)(props, 'badge', 'backgroundColorHoverFocus'));
  }

  return (0, _styledComponents.css)(_templateObject9(), (0, _themeFunctions.getColor)(props, 'badge', 'linkColor'), (0, _themeFunctions.getColor)(props, 'badge', 'backgroundColor'));
};

var Badge = (0, _styledComponents["default"])(_base.Span)(_templateObject10(), function (props) {
  return (0, _themeFunctions.getFontWeight)(props, 'badge', 'default');
}, function (props) {
  return borderRadius(props);
}, function (props) {
  return padding(props);
}, function (props) {
  return color(props);
}, function (props) {
  return backgroundColor(props);
}, function (props) {
  return badgeLink(props);
});
var _default = Badge;
exports["default"] = _default;