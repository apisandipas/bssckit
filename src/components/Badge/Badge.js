import styled, { css } from 'styled-components';
import { theme, ifProp } from 'styled-tools';
import { Span } from '../base';
import defaultTheme from '../../utils/defaultTheme';
import { themeContexts } from '../../utils/consts';

const Badge = styled(Span)`
  display: inline-block;
  padding: ${theme('badge.padding.default')};
  font-size: 75%;
  font-weight: ${theme('badge.fontWeight.default')};
  line-height: 1;

  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: ${theme('badge.borderRadius.default')};

  ${themeContexts.map(context => {
    return ifProp(
      context,
      css`
        color: ${theme(`badge.colors.${context}.color`)};
        background-color: ${theme(`badge.colors.${context}.backgroundColor`)};

        ${ifProp(
          'action',
          css`
            & > a {
              color: ${theme(`badge.colors.${context}.linkColor`)};
              text-decoration: none;
              background-color: ${theme(`badge.colors.${context}.backgroundColor`)};
              &:hover,
              &:focus {
                text-decoration: none;
                background-color: ${theme(`badge.colors.${context}.backgroundColorHoverFocus`)};
              }
            }
            &:hover,
            &:focus {
              cursor: pointer;
              background-color: ${theme(`badge.colors.${context}.backgroundColorHoverFocus`)};
              & > a {
                background-color: ${theme(`badge.colors.${context}.backgroundColorHoverFocus`)};
              }
            }
          `,
          css`
            & > a {
              color: ${theme(`badge.colors.${context}.linkColor`)};
              text-decoration: none;
              background-color: ${theme(`badge.colors.${context}.backgroundColor`)};
            }
          `
        )}
      `
    );
  })}

  ${ifProp(
    'pill',
    css`
      border-radius: ${theme('badge.borderRadius.pill')};
    `
  )}

  /* Empty badges collapse automatically */
  &:empty {
    display: none;
  }
`;

Badge.defaultProps = {
  theme: defaultTheme,
};

export default Badge;
