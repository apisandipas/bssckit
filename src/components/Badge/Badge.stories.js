import React from 'react';
import { Badge } from './index';
import { Button } from '../Button';

export default {
  title: 'Badge',
};

export const inHeadings = () => (
  <>
    <h1>
      Example heading <Badge primary>New</Badge>
    </h1>
    <h2>
      Example heading <Badge primary>New</Badge>
    </h2>
    <h3>
      Example heading <Badge primary>New</Badge>
    </h3>
    <h4>
      Example heading <Badge primary>New</Badge>
    </h4>
    <h5>
      Example heading <Badge primary>New</Badge>
    </h5>
    <h6>
      Example heading <Badge primary>New</Badge>
    </h6>
  </>
);

export const contextual = () => (
  <>
    <Badge primary>Primary</Badge>
    <br />
    <Badge secondary>Secondary</Badge>
    <br />
    <Badge success>Success</Badge>
    <br />
    <Badge danger>Danger</Badge>
    <br />
    <Badge warning>Warning</Badge>
    <br />
    <Badge info>Info</Badge>
    <br />
    <Badge light>Light</Badge>
    <br />
    <Badge dark>Dark</Badge>
  </>
);

export const pills = () => (
  <>
    <Badge pill primary>
      Primary
    </Badge>
    <br />
    <Badge pill secondary>
      Secondary
    </Badge>
    <br />
    <Badge pill success>
      Success
    </Badge>
    <br />
    <Badge pill danger>
      Danger
    </Badge>
    <br />
    <Badge pill warning>
      Warning
    </Badge>
    <br />
    <Badge pill info>
      Info
    </Badge>
    <br />
    <Badge pill light>
      Light
    </Badge>
    <br />
    <Badge pill dark>
      Dark
    </Badge>
  </>
);

export const inButtons = () => (
  <>
    <Button block primary>
      Notifications <Badge light>4</Badge>
    </Button>
    <Button block dark>
      Notifications <Badge light>4</Badge>
    </Button>
    <Button block light>
      Notifications <Badge dark>4</Badge>
    </Button>
  </>
);
