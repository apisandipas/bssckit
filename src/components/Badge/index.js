import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { theme, ifProp } from 'styled-tools';
import { Span } from '../base';

const colors = [
  { name: 'primary', backgroundColor: '#007bff' },
  { name: 'secondary', backgroundColor: '#6c757d' },
  { name: 'success', backgroundColor: '#28a745' },
  { name: 'danger', backgroundColor: '#dc3545' },
  { name: 'warning', backgroundColor: '#ffc107' },
  { name: 'info', backgroundColor: '#17a2b8' },
  { name: 'light', backgroundColor: '#f8f9fa', color: 'rgb(45, 47, 49)' },
  { name: 'dark', backgroundColor: '#343a40' },
];

const Badge = styled(Span)`
  display: inline-block;
  padding: ${theme('badge.paddingX', '0.25rem')} ${theme('badge.paddingY', '0.4rem')};
  font-size: 75%;
  font-weight: ${theme('badge.fontWeight', '700')};
  line-height: 1;
  color: ${theme('badge.fontColor', '#fff')};
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: ${theme('badge.borderRadius', theme('globals.borderRadius', '0.25rem'))};

  ${colors.map(({ name, color, backgroundColor }) => {
    return ifProp(
      name,
      css`
        ${name === 'light' &&
          css`
            color: ${color};
          `}
        background-color: ${backgroundColor};
      `
    );
  })}

  ${ifProp(
    'pill',
    css`
      border-radius: ${theme('badge.pillBorderRadius', theme('global.borderRadius', '10rem'))};
    `
  )}

  & > a {
    color: ${theme('badge.linkColor', 'white')};
    text-decoration: none;
    background-color: ${theme('badge.linkBackgroundColor', 'transparent')};
  }

  /* Empty badges collapse automatically */
  &:empty {
    display: none;
  }
`;

export default Badge;
