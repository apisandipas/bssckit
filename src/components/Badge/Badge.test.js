import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Badge from './index';

test('Styles match plain', () => {
  const tree = renderer.create(<Badge />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'inline-block');
  expect(tree).toHaveStyleRule('padding', '0.25rem 0.4rem');
  expect(tree).toHaveStyleRule('font-size', '75%');
  expect(tree).toHaveStyleRule('font-weight', '700');
  expect(tree).toHaveStyleRule('line-height', '1');
  expect(tree).toHaveStyleRule('text-align', 'center');
  expect(tree).toHaveStyleRule('white-space', 'nowrap');
  expect(tree).toHaveStyleRule('vertical-align', 'baseline');
  expect(tree).toHaveStyleRule('border-radius', '0.25rem');
  expect(tree).toHaveStyleRule('display', 'none', {
    modifier: '&:empty',
  });
});

test('Styles matches primary', () => {
  const tree = renderer.create(<Badge primary />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#007bff');
});

test('Styles matches secondary', () => {
  const tree = renderer.create(<Badge secondary />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#6c757d');
});

test('Styles matches success', () => {
  const tree = renderer.create(<Badge success />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#28a745');
});

test('Styles matches danger', () => {
  const tree = renderer.create(<Badge danger />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#dc3545');
});

test('Styles matches warning', () => {
  const tree = renderer.create(<Badge warning />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#ffc107');
});

test('Styles matches info', () => {
  const tree = renderer.create(<Badge info />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#17a2b8');
});

test('Styles matches light', () => {
  const tree = renderer.create(<Badge light />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', 'rgb(45,47,49)');
  expect(tree).toHaveStyleRule('background-color', '#f8f9fa');
});

test('Styles matches dark', () => {
  const tree = renderer.create(<Badge dark />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background-color', '#343a40');
});
