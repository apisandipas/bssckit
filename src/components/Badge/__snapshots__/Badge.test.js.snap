// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles match plain 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches danger 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #dc3545;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches dark 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #343a40;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches info 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #17a2b8;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches light 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  color: rgb(45,47,49);
  background-color: #f8f9fa;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches primary 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #007bff;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches secondary 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #6c757d;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches success 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #28a745;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;

exports[`Styles matches warning 1`] = `
.c0 {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  background-color: #ffc107;
}

.c0 > a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0:empty {
  display: none;
}

<span
  className="sc-AxirZ c0"
/>
`;
